crates/wasmtime/src/component/linker.rs

```diff
@@ -212,11 +195,59 @@ impl<T> Linker<T> {
             cx.definition(ty, import)
                 .with_context(|| format!("import `{name}` has the wrong type"))?;
         }
+        Ok(cx)
+    }
+
+    /// Iterates over all types imported by this component
+    pub fn import_types<'a>(
```

Comment:
Could this perhaps return a `types::Component`? If so then let's call this method `substituted_component_type` where the documentation indicates that the resource types imported by the `component` specified as an argument are replaced with the imports present in `Linker` in the returned type.

Reply:
Implementing this would probably require adding a `ComponentTypeIndex` somewhere inside of a component since I don't think that exists today. Furthermore that may not be altogether trivial to add. 

If it's not easy to add, some suggestions on this method:

* `ExactSizeIterator` + `&str` return value in the iterator
* Calling this `substituted_import_types` and documenting that the `Linker`'s resources are substituted in the component types. 
* Adding a `substituted_export_types` here as well for symmetry